types/react-query/index.d.ts

```diff
@@ -36,107 +53,218 @@ export interface QueryOptions<TResult> {
     refetchOnWindowFocus?: boolean;
     onError?: (err: any) => void;
     onSuccess?: (data: TResult) => void;
+    onSettled?: (data: TResult, err: any) => void;
     suspense?: boolean;
     initialData?: TResult;
+    refetchOnMount?: boolean;
 }
 
-export interface QueryOptionsPaginated<TResult> extends QueryOptions<TResult> {
-    paginated: true;
-    getCanFetchMore: (lastPage: TResult, allPages: TResult[]) => boolean;
+export interface InfiniteQueryOptions<TResult, FetchMoreVariable> extends QueryOptions<TResult> {
+    getFetchMore?: ((lastGroup: TResult, allGroups: TResult[]) => FetchMoreVariable) | boolean;
```

Comment:
Shouldn't this be?
```suggestion
    getFetchMore?: (lastGroup: TResult, allGroups: TResult[]) => FetchMoreVariable | boolean;
```

From what I understand in the [doc](https://github.com/tannerlinsley/react-query#options-2) and the code it's the result of the function that is either a `FetchMoreVariable` or a `boolean`.

Maybe this could be the right typing:
```ts
    getFetchMore?: (lastGroup: TResult, allGroups: TResult[]) => FetchMoreVariable | false;
```